apps/bluetooth/js/modules/pair_manager.js

```diff
@@ -38,21 +62,176 @@ define(function(require) {
           this.showPendingPairing(event.settingValue);
       }.bind(this));
 
-      // Observe Bluetooth ondisabled event from hardware side.
+      // Observe Bluetooth state from hardware side.
       // Then, close pairing dialog immediately.
-      navigator.mozBluetooth.ondisabled = this.onBluetoothDisabled.bind(this);
+      BtContext.observe('state', (state) => {
+        if (state !== 'enabled') {
+          this.onBluetoothDisabled();
+        }
+      });
 
       // Observe Bluetooth enabled state for close pairing dialog immediately.
-      navigator.mozSettings.addObserver('bluetooth.enabled',
-        function gotBluetoothEnabledChanged(event) {
-          if (!event.settingValue) {
-            this.onBluetoothDisabled();
-          }
-      }.bind(this));
+      navigator.mozSettings.addObserver('bluetooth.enabled', (event) => {
+        if (!event.settingValue) {
+          this.onBluetoothDisabled();
+        }
+      });
+    },
+
+    /**
+     * 'defaultAdapter' change event handler from adapter manager for
+     * updating it immediately.
+     *
+     * @access private
+     * @memberOf PairManager
+     * @param {Object BluetoothAdapter} newAdapter
+     */
+    _onDefaultAdapterChanged: function(newAdapter) {
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+    },
+
+    /**
+     * Watch 'ondisplaypasskeyreq' dom event for pairing.
+     * A handler to trigger when a remote bluetooth device requests to display 
+     * passkey on the screen during pairing process.
+     *
+     * @access private
+     * @memberOf PairManager
+     */
+    _watchOndisplaypasskeyreq: function() {
+      if (!this._defaultAdapter || !this._defaultAdapter.pairingReqs) {
+        return;
+      }
+
+      this._defaultAdapter.pairingReqs.ondisplaypasskeyreq = 
+        this._onDisplayPasskeyReq.bind(this);
+    },
+
+    /**
+     * A handler to handle 'ondisplaypasskeyreq' event while it's coming.
+     *
+     * @access private
+     * @memberOf PairManager
+     */
+    _onDisplayPasskeyReq: function(evt) {
+      Debug('ondisplaypasskeyreq(): Pairing request from ' + 
+            evt.device.name + ': display passkey = ' + evt.handle.passkey);
+      // TODO: Display passkey to user if user story is required.
```

Comment:
Let's throw a exception here, then we are able to be notified from the console when run into this case.